libsolidity/interface/CompilerStack.cpp

```diff
@@ -1113,24 +1113,27 @@ void CompilerStack::compileContract(
 	_otherCompilers[compiledContract.contract] = compiler;
 }
 
-void CompilerStack::generateIR(ContractDefinition const& _contract)
+string CompilerStack::generateIR(ContractDefinition const& _contract)
 {
 	solAssert(m_stackState >= AnalysisPerformed, "");
 	if (m_hasError)
 		BOOST_THROW_EXCEPTION(CompilerError() << errinfo_comment("Called generateIR with errors."));
 
 	if (!_contract.canBeDeployed())
-		return;
+		return "";
 
 	Contract& compiledContract = m_contracts.at(_contract.fullyQualifiedName());
 	if (!compiledContract.yulIR.empty())
-		return;
+		return compiledContract.yulIR;
 
+	string dependenciesSource;
 	for (auto const* dependency: _contract.annotation().contractDependencies)
-		generateIR(*dependency);
+		dependenciesSource += generateIR(*dependency);
```

Comment:
While it looks convenience, it is an inconsistent interface change to return the yulIR from the `generateIR` function. Can you use `m_contracts.at(dependency->fullyQualifiedName).yulIR` to fetch the source?

Reply:
Makes sense, I did fix it now.